cpp/node_module/algorithm/node.cpp

```diff
@@ -1 +1,69 @@
 #include "node.hpp"
+#include <unordered_set>
+
+#include "mgp.hpp"
+
+namespace {
+
+std::unordered_map<std::string_view, uint8_t> get_type_direction(const mgp::Value &types) {
+  std::unordered_map<std::string_view, uint8_t> result;
+  for (const auto &type_value : types.ValueList()) {
+    auto type = type_value.ValueString();
+    if (type.starts_with('<')) {
+      result[type.substr(1, type.size() - 1)] |= 1;
+    } else if (type.ends_with('>')) {
+      result[type.substr(0, type.size() - 1)] |= 2;
+    } else {
+      result[type] |= 3;
+    }
```

Comment:
The case where `type` both starts with `'<'` and ends with `'>'` (e.g. '<KNOWS>') should also be addressed.

Reply:
We internally discussed it and decided that <> case isn't needed because it it is just like without the <> (since it covers both in and out nodes). Will discuss tomorrow on daily to either throw error or allow it.

Reply:
I've checked that and you're right about it :smile:. However, if the user gives e.g. `"<KNOWS>"`, the current code would treat that like `"<KNOWS"` and not `"KNOWS"`.

Reply:
My bad, we already said to throw an error. Fixed